### DISCUSSION TITLE
How to completely delete an app

### DISCUSSION THREAD

**Discussion by junxed** [fail2ban/fail2ban#3591] [Q&A] [answered]
How to properly uninstall an application to reinstall from scratch?

**Comment by sebres** [comment, marked as answer]
It depends on package manager / distribution.
For debians it'd be `apt remove --purge fail2ban` (see https://github.com/fail2ban/fail2ban/issues/3327#issuecomment-1204957099 for details).

If you installed it manually (e. g. from source), it would be enough to remove:
```
?/usr/local?/bin/fail2ban-* /usr/local/lib/python3.*/dist-packages/fail2ban-* /etc/fail2ban /usr/share/doc/fail2ban
```
but better inspect the output of dry-run setup, for instance:
```
python setup.py --dry-run install --without-tests
```
